RELEASES.md

```diff
@@ -0,0 +1,110 @@
+## Release Cadence and Support Policy ##
+
+This document describes the release cadence for runc as well as outlining the
+support policy for old release branches. Historically, despite runc being the
+most widely used Linux container runtime, our release schedule has been very
+ad-hoc and has resulted in very long periods of time between minor releases,
+causing issues for downstreams that wanted particular features.
+
+### Semantic Versioning ###
+
+runc uses [Semantic Versioning][semver] for releases. However, our
+compatibility policy only applies to the runc binary. We will make a
+best-effort attempt to reduce the impact to users that make direct use of the
+Go packages prefixed with `github.com/opencontainers/runc`, but we do not
+formally guarantee that API compatibility will be preserved.
+
+[semver]: https://semver.org/spec/v2.0.0.html
+
+### Release Cadence ###
+
+> **NOTE**: At time of writing, this proposal is still a draft as we have not
+> yet done several releases using this cadence. If you have feedback on this
+> proposal (such as how well this proposal addresses your needs as a downstream
+> packager of runc), please feel free to [open an issue][new-issue] so we can
+> discuss it further. However, the current plan is for this proposal to be
+> followed for the `1.3.0` and `1.4.0` releases in 2025.
+
+[new-issue]: https://github.com/opencontainers/runc/issues/new/choose
+
+runc follows a 6-month minor version release schedule, with the aim of releases
+happening at the end of April and October each year.
+
+The first release candidate will be created 2 months before the planned release
+date (i.e. the end of February and August, respectively), at which point the
+release branch will be created and will enter a feature freeze. No new features
+will be merged into the release branch, and large features being developed
+immediately before the feature freeze may have their merge delayed so as to not
+be included in the next release. Most releases will have two or three release
+candidates, but this may change depending on the circumstances of the release
+at the time.
+
+If a last-minute critical issue is discovered, the release may be delayed.
+However, the following release will still go according to schedule (except in
+the exceptionally unlikely scenario where the delay is 4-6 months long, in
+which case the next release is moved forward to when the subsequent release
+would have been).
+
+Here is a hypothetical release timeline to see how this works in practice:
+
+| Date       | Release      | Notes |
+| ---------- | ------------ | ----- |
+| 200X-02-28 | `1.3.0-rc.1` | `release-1.3` branch created, feature freeze. |
```

Comment:
According to this release policy, the version of `1.3.0-rc.1` will be released in `2025-02-28`. If I understand right, all of tasks in [1.3.0 milestone](https://github.com/opencontainers/runc/milestone/14) should be resolved before we release `1.3.0-rc.1`. There are only about 2.5 months left, I think it's hard to complete it. Maybe we should reorganize the task list for this milestone.

Reply:
If we agree on doing time-based releases then we should move any outstanding items to later release milestones if they don't make the cut. (Also, quite a few of those were added to 1.3 to indicate they were post-1.2 not that they needed to make it into 1.3.)

Reply:
Yes, and we did the same for 1.2 -- moved some PRs/issues to a later release.

With time-based releases, instead of trying to ensure everything that's marked with the next milestone is going in (and delaying a release by a couple of years in the process), we fix/merge what we can, and change the milestone to the next one for the rest. This way we're not perfect (well,we never are) but at least we're predictable wrt when a new release is made.